### ISSUE TITLE
[Bug]: benchmarking type error

### ISSUE THREAD

**Issue by yeslogin2** [openvinotoolkit/anomalib#1149]
### Describe the bug

Hi,

Following the docs of benchmarking, the simple command of "`python tools/benchmarking/benchmark.py --config tools/benchmarking/benchmark_params.yaml`" produces unexpected results.

![2023-06-24_19-22](https://github.com/openvinotoolkit/anomalib/assets/17853673/ff90dbdd-37a4-487c-bfe5-1fcdf7f4c642)

An exception of TypeError was thrown out indicating that the path value was invalid.

When checking the code in benchmark.py, line 114, I found this:

![2023-06-24_19-09](https://github.com/openvinotoolkit/anomalib/assets/17853673/7fb3418a-d42f-43a2-91ce-3a772c606475)

passing `model_config` which is a dict, to `get_torch_throughput` which expects a path.

I reckon this is a bug if I am not wrong in somewhere.

Thank you if you can help to verify and fix this.

### Dataset

N/A

### Model

N/A

### Steps to reproduce the behavior

1. Install
2. Prepare datasets
3. Run command "python tools/benchmarking/benchmark.py --config tools/benchmarking/benchmark_params.yaml"

### OS information

OS information:
- OS: [e.g. Ubuntu 20.04]
- Python version: [e.g. 3.8.10]
- Anomalib version: [e.g. 0.6]
- PyTorch version: [e.g. 1.9.0]
- CUDA/cuDNN version: [e.g. 11.1]
- GPU models and configuration: [e.g. 2x GeForce RTX 3050]

### Expected behavior

Should run benchmarks normally and no TypeErrors

### Pip/GitHub

pip

### Configuration YAML

```yaml
seed: 42
compute_openvino: false
hardware:
  # - cpu
  - gpu
writer:
  - wandb
  - tensorboard
dataset:
    category:
      - bottle
      - cable
      - capsule
      - carpet
      - grid
      - hazelnut
      - leather
      - metal_nut
      - pill
      - screw
      - tile
      - toothbrush
      - transistor
      - wood
      - zipper
    image_size: [224]
  model_name:
    - padim
    - patchcore
```

### Logs

```shell
Traceback (most recent call last):
  File "/home/young/ppp/py_learn/IndustrailAnomalyDetection/anomalib/tools/benchmarking/benchmark.py", line 69, in wrapper
    value = func(*args, **kwargs)
  File "/home/young/ppp/py_learn/IndustrailAnomalyDetection/anomalib/tools/benchmarking/benchmark.py", line 114, in get_single_model_metrics
    throughput = get_torch_throughput(model_config, model, datamodule.test_dataloader().dataset)
  File "/home/young/ppp/py_learn/IndustrailAnomalyDetection/anomalib/src/anomalib/utils/sweep/helpers/inference.py", line 29, in get_torch_throughput
    model_path = Path(model_path)
  File "/usr/lib/python3.8/pathlib.py", line 1042, in __new__
    self = cls._from_parts(args, init=False)
  File "/usr/lib/python3.8/pathlib.py", line 683, in _from_parts
    drv, root, parts = self._parse_args(args)
  File "/usr/lib/python3.8/pathlib.py", line 667, in _parse_args
    a = os.fspath(a)
TypeError: expected str, bytes or os.PathLike object, not DictConfig
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by blaz-r** [comment]
Hello. I'm trying to reproduce this right now. I'll get back to you asap.

**Comment by blaz-r** [comment]
@yeslogin2 This should now be fixed with PR #1155.
